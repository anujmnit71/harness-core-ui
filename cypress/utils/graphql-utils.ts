export const hasOperationName: (req: any, operationName: string) => boolean = (req, operationName) => {
  const { body } = req
  // eslint-disable-next-line no-prototype-builtins
  return body.hasOwnProperty('operationName') && body.operationName === operationName
}

export const aliasQuery: (req: any, operationName: string) => void = (req, operationName) => {
  if (hasOperationName(req, operationName)) {
    req.alias = `gql${operationName}Query`
  }
}
