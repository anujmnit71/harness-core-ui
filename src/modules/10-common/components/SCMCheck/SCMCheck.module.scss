/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.userInfo {
  position: absolute;
  top: var(--spacing-medium);
  right: var(--spacing-huge);
}

.addUserContainer {
  background: var(--orange-200) !important;
  padding: var(--spacing-medium) var(--spacing-large) !important;
}
