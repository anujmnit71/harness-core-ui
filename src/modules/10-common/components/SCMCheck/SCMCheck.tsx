/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useEffect } from 'react'
import { Link, useParams } from 'react-router-dom'
import { Avatar, Color, Container, FontVariation, Icon, Layout, Text } from '@harness/uicore'
import { noop, defaultTo } from 'lodash-es'
import cx from 'classnames'
import { useGitSyncStore } from 'framework/GitRepoStore/GitSyncStoreContext'
import paths from '@common/RouteDefinitions'
import { useStrings } from 'framework/strings/String'
import type { AccountPathProps } from '@common/interfaces/RouteInterfaces'
import css from './SCMCheck.module.scss'

export interface SCMCheckProps {
  profileLinkClickHandler?: () => void // If used inside modal, this can be use to close modal
  title?: string
  className?: string
  validateSCM?: (hasValidSCM: boolean) => void // Can be used to disable the commit
}

const SCMCheck: React.FC<SCMCheckProps> = props => {
  const { codeManagers, loadingCodeManagers } = useGitSyncStore()
  const { accountId } = useParams<AccountPathProps>()
  const [currentUserInfo, setCurrentUserInfo] = React.useState('')
  const { getString } = useStrings()
  const { validateSCM = noop, profileLinkClickHandler = noop, className = '', title } = props

  const getUserDisplayName = (user?: string): string => {
    return defaultTo(user, getString('na'))
  }

  useEffect(() => {
    const userSpec = codeManagers?.[0]?.authentication?.spec?.spec
    const user = userSpec?.username || userSpec?.usernameRef

    setCurrentUserInfo(user)
    validateSCM(!loadingCodeManagers && user)
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [codeManagers])

  return (
    <Container className={cx(className)} margin={{ bottom: 'large' }}>
      {currentUserInfo ? (
        <Layout.Horizontal className={css.userInfo} flex={{ alignItems: 'center' }}>
          <Avatar
            size="small"
            name={getUserDisplayName(currentUserInfo)}
            backgroundColor={Color.PRIMARY_7}
            hoverCard={false}
          />
          <Text color={Color.GREY_700}>
            {getString('common.git.currentUserLabel', { user: getUserDisplayName(currentUserInfo) })}
          </Text>
        </Layout.Horizontal>
      ) : (
        <></>
      )}
      {title ? (
        <Text font={{ variation: FontVariation.H3 }} margin={{ bottom: 'large', top: 'small' }}>
          {title}
        </Text>
      ) : (
        <></>
      )}

      {!loadingCodeManagers && !currentUserInfo && (
        <Layout.Horizontal className={css.addUserContainer} spacing="small">
          <Icon name="warning-sign" color={Color.ORANGE_700}></Icon>
          <div>
            <Text font={{ variation: FontVariation.SMALL }} color={Color.BLACK}>
              {getString('common.git.noUserLabel')}
            </Text>
            <Link to={paths.toUserProfile({ accountId })} target="_blank" onClick={profileLinkClickHandler}>
              <Text inline font={{ variation: FontVariation.SMALL_BOLD }} color={Color.PRIMARY_7}>
                {getString('common.git.addUserCredentialLabel')}
              </Text>
            </Link>
          </div>
        </Layout.Horizontal>
      )}
    </Container>
  )
}

export default SCMCheck
