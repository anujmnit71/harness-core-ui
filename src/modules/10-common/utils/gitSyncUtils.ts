/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { GitSyncConfig } from 'services/cd-ng'
import { folderPathName, yamlFileExtension } from './StringUtils'

export const getRepoDetailsByIndentifier = (identifier: string | undefined, repos: GitSyncConfig[]) => {
  return repos.find((repo: GitSyncConfig) => repo.identifier === identifier)
}

export const validateFilePath = (filePath: string): boolean => {
  if (!filePath.endsWith(yamlFileExtension)) {
    return false
  } else {
    const fullPath = filePath.slice(0, filePath.length - yamlFileExtension.length)
    const subPaths = fullPath.split('/')
    return subPaths.every((folderName: string) => {
      return folderName.length && folderName.match(folderPathName)?.length
    })
  }
}
