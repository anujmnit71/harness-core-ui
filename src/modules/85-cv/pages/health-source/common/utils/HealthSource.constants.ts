/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export const HealthSourceFieldNames = {
  MANUAL_QUERY: 'isManualQuery',
  IDENTIFIER: 'identifier',
  METRIC_NAME: 'metricName',
  METRIC_DATA: 'metricData',
  GROUP_NAME: 'groupName',
  CONTINUOUS_VERIFICATION: 'continuousVerification',
  HEALTH_SCORE: 'healthScore',
  SLI: 'sli',
  RISK_CATEGORY: 'riskCategory',
  HIGHER_BASELINE_DEVIATION: 'higherBaselineDeviation',
  LOWER_BASELINE_DEVIATION: 'lowerBaselineDeviation'
}
