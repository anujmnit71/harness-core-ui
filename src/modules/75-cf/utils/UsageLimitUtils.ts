/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { ReactNode } from 'react'
import { BannerType } from '@common/layouts/Constants'
import type { CheckFeatureReturn } from 'framework/featureStore/featureStoreUtil'
import { formatToCompactNumber } from '@cf/utils/CFUtils'
import type { UseStringsReturn } from 'framework/strings'

export const getBannerText = (
  getString: UseStringsReturn['getString'],
  monthlyActiveUsers: CheckFeatureReturn,
  additionalLicenseProps: Record<string, any>
): ReactNode => {
  const { isEnterpriseEdition, isFreeEdition, isTeamEdition } = additionalLicenseProps

  const clientMauUsageCount = Number(monthlyActiveUsers?.featureDetail?.count)
  const clientMauPlanLimit = Number(monthlyActiveUsers?.featureDetail?.limit)

  const clientMauUsagePercentage = Math.trunc((clientMauUsageCount / clientMauPlanLimit) * 100)
  const clientMauPlanLimitFormatted = formatToCompactNumber(clientMauPlanLimit)

  const showInfoBanner = clientMauPlanLimit > 0 && clientMauUsagePercentage >= 90 && clientMauUsagePercentage < 100
  const showWarningBanner = clientMauPlanLimit > 0 && clientMauUsagePercentage >= 100

  let message!: string
  let bannerType!: BannerType

  if (isFreeEdition) {
    if (!showInfoBanner) {
      return {
        message: () =>
          getString('cf.planEnforcement.freePlan.approachingLimit', {
            clientMauUsagePercentage,
            clientMauPlanLimitFormatted
          }),
        bannerType: BannerType.INFO
      }
    }

    if (showWarningBanner) {
      return {
        message: getString('cf.planEnforcement.freePlan.upgradeRequired', {
          clientMauPlanLimitFormatted
        }),
        bannerType: BannerType.LEVEL_UP
      }
    }
  }

  if (isTeamEdition || isEnterpriseEdition) {
    if (showInfoBanner) {
      return {
        message: () =>
          getString('cf.planEnforcement.teamEnterprisePlan.approachingLimit', {
            clientMauUsagePercentage
          }),
        bannerType: BannerType.INFO
      }
    }

    if (showWarningBanner) {
      return {
        message: () => getString('cf.planEnforcement.teamEnterprisePlan.upgradeRequired'),
        bannerType: BannerType.LEVEL_UP
      }
    }
  }

  return {
    message,
    bannerType
  }
}
