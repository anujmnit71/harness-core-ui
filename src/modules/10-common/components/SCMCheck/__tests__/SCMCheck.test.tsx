/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, waitFor } from '@testing-library/react'
import { noop } from 'lodash-es'
import { GitSyncTestWrapper } from '@common/utils/gitSyncTestUtils'
import SCMCheck from '../SCMCheck'

const pathParams = { accountId: 'dummy', orgIdentifier: 'default', projectIdentifier: 'dummyProject' }
const title = 'Save to git'
const handlerForSCM = jest.fn()

describe('Test SCMCheck', () => {
  test('SCMCheck while loading should render none of user info or missing SCM warning', async () => {
    const { container, getByText } = render(
      <GitSyncTestWrapper
        path="/account/:accountId/ci/orgs/:orgIdentifier/projects/:projectIdentifier/admin/git-sync/repos"
        pathParams={pathParams}
        gitSyncStoreValues={{ loadingCodeManagers: true, codeManagers: [] }}
      >
        <SCMCheck title={title} validateSCM={noop} />
      </GitSyncTestWrapper>
    )
    expect(container).toMatchSnapshot()
    await waitFor(() => {
      expect(getByText(title)).toBeInTheDocument()
    })
  })

  test('SCMCheck should render currentUserLabel if SCM is present', async () => {
    const { container, getByText } = render(
      <GitSyncTestWrapper
        path="/account/:accountId/ci/orgs/:orgIdentifier/projects/:projectIdentifier/admin/git-sync/repos"
        pathParams={pathParams}
        defaultAppStoreValues={{ currentUserInfo: { name: 'dev', email: 'mail@harness.io' } }}
      >
        <SCMCheck title={title} validateSCM={handlerForSCM} />
      </GitSyncTestWrapper>
    )
    expect(container).toMatchSnapshot()
    await waitFor(() => {
      expect(getByText('common.git.currentUserLabel')).toBeInTheDocument()
    })
    expect(handlerForSCM).toBeCalled()
  })

  test('SCMCheck should render currentUserLabel using email if name is not present', async () => {
    const { container } = render(
      <GitSyncTestWrapper
        path="/account/:accountId/ci/orgs/:orgIdentifier/projects/:projectIdentifier/admin/git-sync/repos"
        pathParams={pathParams}
        defaultAppStoreValues={{ currentUserInfo: { email: 'mail@harness.io' } }}
      >
        <SCMCheck title={title} validateSCM={handlerForSCM} />
      </GitSyncTestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('SCMCheck should render warning for missing SCM with link to addUserCredentialLabel', async () => {
    const { container, getByText } = render(
      <GitSyncTestWrapper
        path="/account/:accountId/ci/orgs/:orgIdentifier/projects/:projectIdentifier/admin/git-sync/repos"
        pathParams={pathParams}
        gitSyncStoreValues={{ loadingCodeManagers: false, codeManagers: [] }}
      >
        <SCMCheck title={title} validateSCM={handlerForSCM} />
      </GitSyncTestWrapper>
    )
    expect(container).toMatchSnapshot()
    await waitFor(() => {
      expect(getByText('common.git.noUserLabel')).toBeInTheDocument()
      expect(getByText('common.git.addUserCredentialLabel')).toBeInTheDocument()
    })
    expect(handlerForSCM).toBeCalled()
  })
})
