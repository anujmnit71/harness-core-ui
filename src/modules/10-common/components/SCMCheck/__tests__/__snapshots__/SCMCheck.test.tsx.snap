// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SCMCheck SCMCheck should render currentUserLabel if SCM is present 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--margin-bottom-large"
  >
    <div
      class="Layout--horizontal StyledProps--main userInfo StyledProps--flex StyledProps--flex-alignItems-center"
    >
      <div
        class="Avatar--Avatar Avatar--contentStyle"
      >
        <span
          class="bp3-popover-wrapper Avatar--avatarPopOver"
        >
          <span
            class="bp3-popover-target"
          >
            <div
              class="Avatar--AvatarInner"
              style="border-radius: 100%; line-height: 18px; width: 24px; height: 24px; font-size: 10px;"
              tabindex="0"
            >
              D
            </div>
          </span>
        </span>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700"
      >
        common.git.currentUserLabel
      </p>
    </div>
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-variation-h3 StyledProps--margin-bottom-large StyledProps--margin-top-small"
    >
      Save to git
    </p>
  </div>
</div>
`;

exports[`Test SCMCheck SCMCheck should render currentUserLabel using email if name is not present 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--margin-bottom-large"
  >
    <div
      class="Layout--horizontal StyledProps--main userInfo StyledProps--flex StyledProps--flex-alignItems-center"
    >
      <div
        class="Avatar--Avatar Avatar--contentStyle"
      >
        <span
          class="bp3-popover-wrapper Avatar--avatarPopOver"
        >
          <span
            class="bp3-popover-target"
          >
            <div
              class="Avatar--AvatarInner"
              style="border-radius: 100%; line-height: 18px; width: 24px; height: 24px; font-size: 10px;"
              tabindex="0"
            >
              D
            </div>
          </span>
        </span>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700"
      >
        common.git.currentUserLabel
      </p>
    </div>
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-variation-h3 StyledProps--margin-bottom-large StyledProps--margin-top-small"
    >
      Save to git
    </p>
  </div>
</div>
`;

exports[`Test SCMCheck SCMCheck should render warning for missing SCM with link to addUserCredentialLabel 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--margin-bottom-large"
  >
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-variation-h3 StyledProps--margin-bottom-large StyledProps--margin-top-small"
    >
      Save to git
    </p>
    <div
      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main addUserContainer"
    >
      <span
        class="bp3-icon bp3-icon-warning-sign StyledProps--main StyledProps--color StyledProps--color-orange700"
        icon="warning-sign"
      >
        <svg
          data-icon="warning-sign"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            warning-sign
          </desc>
          <path
            d="M15.84 13.5l.01-.01-7-12-.01.01c-.17-.3-.48-.5-.85-.5s-.67.2-.85.5l-.01-.01-7 12 .01.01c-.09.15-.15.31-.15.5 0 .55.45 1 1 1h14c.55 0 1-.45 1-1 0-.19-.06-.35-.15-.5zm-6.85-.51h-2v-2h2v2zm0-3h-2v-5h2v5z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <div>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-black"
        >
          common.git.noUserLabel
        </p>
        <a
          href="/account/dummy/user/profile"
          target="_blank"
        >
          <span
            class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--font-variation-small-bold StyledProps--color StyledProps--color-primary7"
          >
            common.git.addUserCredentialLabel
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test SCMCheck SCMCheck while loading should render none of user info or missing SCM warning 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main StyledProps--margin-bottom-large"
  >
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-variation-h3 StyledProps--margin-bottom-large StyledProps--margin-top-small"
    >
      Save to git
    </p>
  </div>
</div>
`;
